debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java

```diff
@@ -191,7 +198,9 @@ else if (message.getOperation() == Operation.COMMIT) {
                         pauseNoMessage.sleepWhen(true);
                     }
                 }
-                connection.commit();
+                if (!isInPreSnapshotCatchUpStreaming()) {
```

Comment:
Why is the conditional here?

Reply:
Related to the above comments but we need to hold a transaction open through the catch up streaming so that we can stream up a specific lsn and not lose the view of data at the transaction opened at that location. Since we need to hold the transaction open for the snapshot, this block must not commit during catch up streaming.

Reply:
Ok, gotcha. So as suggested above, having one comment somewhere in the code about the general approach for TX handling/lifecycle would be appreciated.

Reply:
Addressed this comment on #1627